File: Foody/app/src/main/java/prateek_gupta/foody/data/database/RecipesEntity.java
package prateek_gupta.foody.data.database;

import androidx.room.Entity;
import androidx.room.PrimaryKey;

import prateek_gupta.foody.models.FoodRecipe;
import prateek_gupta.foody.util.Constants;

@Entity(tableName=Constants.RECIPES_TABLE)
public class RecipesEntity {
    @PrimaryKey(autoGenerate = false)
    Integer id=0;

    FoodRecipe foodRecipe;

    public RecipesEntity(FoodRecipe foodRecipe) {
        this.foodRecipe = foodRecipe;
    }

    public FoodRecipe getFoodRecipe() {
        return foodRecipe;
    }

    public void setFoodRecipe(FoodRecipe foodRecipe) {
        this.foodRecipe = foodRecipe;
    }
}

File: Foody/app/src/main/java/prateek_gupta/foody/data/database/entities/RecipesEntity.kt
package prateek_gupta.foody.data.database.entities

import androidx.room.Entity
import androidx.room.PrimaryKey
import prateek_gupta.foody.models.FoodRecipe
import prateek_gupta.foody.util.Constants.Companion.RECIPES_TABLE

@Entity(tableName = RECIPES_TABLE)
class RecipesEntity(
    var foodRecipe: FoodRecipe
) {
    @PrimaryKey(autoGenerate = false)
    var id:Int=0
}